debugging/coreclr/appStorage.ts

```diff
@@ -0,0 +1,63 @@
+/*---------------------------------------------------------
+ * Copyright (C) Microsoft Corporation. All rights reserved.
+ *--------------------------------------------------------*/
+
+import * as path from 'path';
```

Comment:
We need to discuss this.

Reply:
As discussed in a comment with @EricJizbaMSFT, while VS Code has a memento service for data associated with the user as well as with a given workspace, it doesn't have something (that I know of) that is associated with an arbitrary folder (which may be one of many subfolders within a workspace and, in the case of debugging, corresponds to a given application being debugged).  Consider, for example, opening VS Code to the solution-level folder that has several applications, each of which could be debugged individually; we want to cache some data for each individual application.

I could likely come up with a key-naming scheme to stuff this data in the workspace memento service, but it just feels like the data should belong to each application itself (as well as is consistent with where this data is stored in VS).